File: yangxj96-serve/yangxj96-serve-dept/src/main/java/io/github/yangxj96/server/dept/controller/DemoController.java
package io.github.yangxj96.server.dept.controller;

import io.github.yangxj96.starter.remote.clients.DemoFeignClient;
import jakarta.annotation.Resource;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/demo")
public class DemoController {

    @Resource
    private DemoFeignClient demoFeignClient;

    @GetMapping("/d1")
    public String d1() {
        return demoFeignClient.get();
    }

}

File: yangxj96-serve/yangxj96-serve-auth/src/main/java/com/yangxj96/saas/server/auth/controller/DemoController.kt
package com.yangxj96.saas.server.auth.controller

import com.yangxj96.saas.bean.system.Route
import com.yangxj96.saas.starter.remote.clients.PlatformFeignClient
import jakarta.annotation.Resource
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/demo")
class DemoController {

    @Resource
    private lateinit var platformFeignClient: PlatformFeignClient

    @GetMapping("/d1")
    fun d1(): MutableList<Route> {
        return platformFeignClient.getAllRoute()
    }

}